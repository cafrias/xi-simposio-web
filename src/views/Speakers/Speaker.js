import React from 'react'

import './Speaker.sass'

function Speaker({pic, name, inst}) {
  return (
    <div className="speaker">
      <img className="speaker__pic" src={pic} alt={`${name} portrait`}/>
      <h2 className="speaker__name">{name}</h2>
      <p className="speaker__inst">{inst}</p>
    </div>
  )
}

export default Speaker
