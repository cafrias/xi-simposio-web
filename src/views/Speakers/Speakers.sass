@import '../../_sass/_breakpoints.sass'

.speakers
  width: 100%

.speakers__grid
  display: flex
  flex-direction: column
  justify-content: center
  padding-top: 2.618rem
  width: 100%

+for-tablet-portrait-up
  .speakers__grid
    flex-direction: row
    flex-wrap: wrap
    justify-content: space-between

+for-desktop-up
  .speakers__grid
    width: 75%
