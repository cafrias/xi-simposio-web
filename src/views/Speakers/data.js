export default [
  {
    pic: 'http://lorempixel.com/200/200/people',
    name: 'Ricardo Barrera',
    inst: '(UNTDF-UNPSJB)'
  },
  {
    pic: 'http://lorempixel.com/200/200/people',
    name: 'Enrique Fowler Newton',
    inst: '(UBA)'
  },
  {
    pic: 'http://lorempixel.com/200/200/people',
    name: 'Juan Permuy',
    inst: '(UBA)'
  },
  {
    pic: 'http://lorempixel.com/200/200/people',
    name: 'Ana MariÃÅa Campo',
    inst: '(UBA)'
  },
  {
    pic: 'http://lorempixel.com/200/200/people',
    name: 'Francisco Provenzani',
    inst: '(UBA)'
  },
  {
    pic: 'http://lorempixel.com/200/200/people',
    name: 'Oscar Botino',
    inst: '(UBA)'
  }
]
