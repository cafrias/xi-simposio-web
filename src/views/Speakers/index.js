// @flow

import React, {Component} from 'react'

import './Speakers.sass'

// COMPONENTS __________________________________________________________________
import Section from '../../components/Section'
import Speaker from './Speaker'

// DATA ________________________________________________________________________
import speakers from './data'

class Speakers extends Component {
  state: {
    shouldAppear: boolean
  }

  constructor(props: any) {
    super(props)

    this.state = {
      shouldAppear: false
    }
  }

  componentDidMount() {
    window.addEventListener('scroll', (e) => {
      const should = this.refs._grid.getBoundingClientRect().top < 250

      if(should && !this.state.shouldAppear) {
        console.log('triggered')
        this.setState({
          shouldAppear: true
        })
      }
    })
  }

  render() {
    const {
      shouldAppear
    } = this.state
    return (
      <Section id="speakers" name="Disertantes" className="background_orange speakers">
        <div ref="_grid"
          className={`speakers__grid ${shouldAppear ? 'speakers__grid--mount' : ''}`}>
          {
            speakers.map((speaker, i) => (
              <Speaker key={i} {...speaker}/>
            ))
          }
        </div>
      </Section>
    )
  }
}

export default Speakers
