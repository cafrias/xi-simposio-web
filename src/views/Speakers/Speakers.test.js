import React from 'react'
import {shallow} from 'enzyme'

import Speakers from './index'

describe('Speakers view', () => {
  var wrapper

  beforeEach(() => wrapper = shallow(<Speakers />))

  it('should render without errors', () => {
    expect(wrapper).toHaveLength(1)
  })

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot()
  })
})
