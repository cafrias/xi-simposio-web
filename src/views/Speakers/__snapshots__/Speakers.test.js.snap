exports[`Speakers view should match snapshot 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <Section
    className="background_orange speakers"
    id="speakers"
    name="Disertantes">
    <div
        className="speakers__grid speakers__grid--mount">
        <Speaker
            id="01"
            inst="(UBA)"
            name="Ricardo Pahlen Acuña"
            pic="01.jpg" />
        <Speaker
            id="02"
            inst="(UBA)"
            name="Ana María Campo"
            pic="02.jpg" />
        <Speaker
            id="04"
            inst="(UNTDF-UNPSJB)"
            name="Ricardo Barrera"
            pic="04.jpg" />
        <Speaker
            id="05"
            inst="(UBA)"
            name="Francisco Provenzani"
            pic="05.jpg" />
        <Speaker
            id="06"
            inst="(UBA)"
            name="Hugo Bottino"
            pic="06.jpg" />
        <Speaker
            id="07"
            inst="(UBA)"
            name="Juan Permuy"
            pic="07.jpg" />
        <Speaker
            id="08"
            inst="(UBA)"
            name="Mauricio Lambertucci"
            pic="08.jpg" />
    </div>
</Section>,
  "nodes": Array [
    <Section
      className="background_orange speakers"
      id="speakers"
      name="Disertantes">
      <div
            className="speakers__grid speakers__grid--mount">
            <Speaker
                  id="01"
                  inst="(UBA)"
                  name="Ricardo Pahlen Acuña"
                  pic="01.jpg" />
            <Speaker
                  id="02"
                  inst="(UBA)"
                  name="Ana María Campo"
                  pic="02.jpg" />
            <Speaker
                  id="04"
                  inst="(UNTDF-UNPSJB)"
                  name="Ricardo Barrera"
                  pic="04.jpg" />
            <Speaker
                  id="05"
                  inst="(UBA)"
                  name="Francisco Provenzani"
                  pic="05.jpg" />
            <Speaker
                  id="06"
                  inst="(UBA)"
                  name="Hugo Bottino"
                  pic="06.jpg" />
            <Speaker
                  id="07"
                  inst="(UBA)"
                  name="Juan Permuy"
                  pic="07.jpg" />
            <Speaker
                  id="08"
                  inst="(UBA)"
                  name="Mauricio Lambertucci"
                  pic="08.jpg" />
      </div>
</Section>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <Speakers />,
      "_debugID": 3,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 2,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <Section
          className="background_orange speakers"
          id="speakers"
          name="Disertantes">
          <div
                    className="speakers__grid speakers__grid--mount">
                    <Speaker
                              id="01"
                              inst="(UBA)"
                              name="Ricardo Pahlen Acuña"
                              pic="01.jpg" />
                    <Speaker
                              id="02"
                              inst="(UBA)"
                              name="Ana María Campo"
                              pic="02.jpg" />
                    <Speaker
                              id="04"
                              inst="(UNTDF-UNPSJB)"
                              name="Ricardo Barrera"
                              pic="04.jpg" />
                    <Speaker
                              id="05"
                              inst="(UBA)"
                              name="Francisco Provenzani"
                              pic="05.jpg" />
                    <Speaker
                              id="06"
                              inst="(UBA)"
                              name="Hugo Bottino"
                              pic="06.jpg" />
                    <Speaker
                              id="07"
                              inst="(UBA)"
                              name="Juan Permuy"
                              pic="07.jpg" />
                    <Speaker
                              id="08"
                              inst="(UBA)"
                              name="Mauricio Lambertucci"
                              pic="08.jpg" />
          </div>
</Section>,
        "_debugID": 4,
        "_renderedOutput": <Section
          className="background_orange speakers"
          id="speakers"
          name="Disertantes">
          <div
                    className="speakers__grid speakers__grid--mount">
                    <Speaker
                              id="01"
                              inst="(UBA)"
                              name="Ricardo Pahlen Acuña"
                              pic="01.jpg" />
                    <Speaker
                              id="02"
                              inst="(UBA)"
                              name="Ana María Campo"
                              pic="02.jpg" />
                    <Speaker
                              id="04"
                              inst="(UNTDF-UNPSJB)"
                              name="Ricardo Barrera"
                              pic="04.jpg" />
                    <Speaker
                              id="05"
                              inst="(UBA)"
                              name="Francisco Provenzani"
                              pic="05.jpg" />
                    <Speaker
                              id="06"
                              inst="(UBA)"
                              name="Hugo Bottino"
                              pic="06.jpg" />
                    <Speaker
                              id="07"
                              inst="(UBA)"
                              name="Juan Permuy"
                              pic="07.jpg" />
                    <Speaker
                              id="08"
                              inst="(UBA)"
                              name="Mauricio Lambertucci"
                              pic="08.jpg" />
          </div>
</Section>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <Speakers />,
}
`;
